integration-tests/src/lib.rs

```diff
@@ -190,3 +191,21 @@ fn testing_oidc() -> AuthenticatorConfig {
         },
     }
 }
+
+pub async fn get_response(port: u16, api_endpoint: &str, exp_status: reqwest::StatusCode, context: &ProviderContext)-> Option<Value>{
+    let url = format!("http://localhost:{}/{}",port,api_endpoint);
+    let response = reqwest::Client::new()
+        .get(&url)
+        .inject_token(&context.provider_manager)
+        .await
+        .unwrap()
+        .send()
+        .await
+        .unwrap();
+    assert_eq!(exp_status,response.status(), "Expected response code does not match with actual response");
+    if exp_status == StatusCode::BAD_REQUEST{
+        None
+    }else{
```

Comment:
The fmt CI test will likely complain about the lack of spaces here. I think every editor/ide has some way to integrate `cargo fmt` into your workflow.

Reply:
Added the block under settings.json on vscode for formatter `{
    "editor.defaultFormatter": "rust-lang.rust-analyzer",
    "editor.formatOnSave": true
}`